### PR TITLE
Add support for integration with oVirt CI System

### PR DESCRIPTION
**Pull request by imjoey**
### Description of the Change

By referring to the documentation of integrating with oVirt CI System [1][2], the automation and packaging directory are both added accordingly.

[1]: https://ovirt-infra-docs.readthedocs.io/en/latest/CI/Build_and_test_standards/index.html
[2]: https://ovirt-infra-docs.readthedocs.io/en/latest/CI/Using_STDCI_with_GitHub/index.html

Signed-off-by: imjoey <majunjiev@gmail.com>

### CODE REVIEW

File: automation/build.packages.el7 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+git
+java-1.8.0-openjdk-devel
+maven
+golang-1.11.2-3
+rpm-build
+rpmdevtools
```
Comment (sandrobonazzola) [inline automation/build.packages.el7]: missing new line at end of file

File: automation/build.repos [hunk 2]
```diff
@@ -0,0 +1 @@
+epel-el7,https://dl.fedoraproject.org/pub/epel/7/x86_64/
```
Comment (sandrobonazzola) [inline automation/build.repos]: missing new line at end of file

File: packaging/spec.el7.in [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+%install
+
+
+%files
+%doc README.adoc
+%doc examples
+%license LICENSE.txt
+
+%changelog
```
Comment (sandrobonazzola) [inline packaging/spec.el7.in]: Please add a changelog line here for 4.3.0-2

**Comment by sandrobonazzola** [comment]
ci test please

File: packaging/spec.el7.in [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+This package contains the Go SDK for version 4 of the oVirt Engine
+API.
+
+%prep
+%setup -q -n ovirt-engine-sdk-go-%{tar_version}
+
+%build
+
+
+%install
```
Comment (machacekondra) [inline packaging/spec.el7.in]: Will you try to setup packaging as per: https://fedoraproject.org/wiki/More_Go_packaging#Walkthrough ?

**Comment by sandrobonazzola** [comment]
Also missing an automation file like https://github.com/oVirt/ovirt-ansible/blob/master/automation.yaml

**Comment by sandrobonazzola** [comment]
ci test please

File: automation/build.packages.el7 (hunk 1 above)
Comment (imjoey) [reply automation/build.packages.el7]: Done, thanks.

File: automation/build.repos (hunk 2 above)
Comment (imjoey) [reply automation/build.repos]: Done, thanks.

File: packaging/spec.el7.in (hunk 3 above)
Comment (imjoey) [reply packaging/spec.el7.in]: Done, thanks.

File: packaging/spec.el7.in (hunk 4 above)
Comment (imjoey) [reply packaging/spec.el7.in]: Thanks Ondra. I've updated the packaging according to the doc. IMHO, the doc seems to be not well applied to this project, as this is not a pure go project hosted in github. After downloading the generator from github, we also need to run `mvn package` to generate the eventual go code, followed by building devel package and binaries as mentioned in the doc. This job has not been finished yet and I'll continue working on this.

Comment (machacekondra) [reply packaging/spec.el7.in]: Yeah, in the other generated SDKs, we just pass a tarball with already generated code. I think we can do it same way here?

**Comment by imjoey** [comment]
Hi @sandrobonazzola , shall we have a discussion about the transfer of the `ovirt/go-ovirt`? 

Since the paths defined in examples and README for `go-ovirt` package are still as `github.com/imjoey/go-ovirt`(See #143 for details.), I have not tagged the project `ovirt/ovirt-engine-sdk-go` with `4.3.0` for now.  From my perspective, the tag `4.3.0` of `ovirt/ovirt-engine-sdk-go` could be committed only after we could use import the go sdk via `import "github.com/ovirt/go-ovirt"`. What's your insights on this, please let me know. Thank you very much.

File: packaging/spec.el7.in (hunk 4 above)
Comment (imjoey) [reply packaging/spec.el7.in]: Yes, gotcha! I'll dig into the `run.py` in `automation/` and implement it on that way. Thanks.

Comment (machacekondra) [reply packaging/spec.el7.in]: Sure, no problem with that. Thanks for doing this.

**Comment by imjoey** [comment]
Hi @sandrobonazzola , the CI testing has passed for `el7`. While could you do me a favor to help me with the `fc28` CI testing failure?  As shown in [here](https://jenkins.ovirt.org/blue/organizations/jenkins/oVirt_ovirt-engine-sdk-go_standard-check-pr/detail/oVirt_ovirt-engine-sdk-go_standard-check-pr/7/pipeline#step-207-log-193). I've added `%undefine _debugsource_packages` and `%undefine _debuginfo_subpackages` at the head of `spec.fc28`, but this could not fix it. Thanks in advance.

File: packaging/spec.el7.in [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+go test -v ./
+rm -fr %{build_gopath}/src/github.com/stretchr
+rm -fr %{build_gopath}/pkg/*
+
+
+%clean
+rm -rf %{buildroot}
+
+
+%files devel
```
[the remark this answers is not in the document]
Comment (machacekondra) [reply packaging/spec.el7.in]: But of course, if you think -devel is better you will get my +1 :)

**Comment by machacekondra** [comment]
@imjoey It complains because you miss %files, you have only %files devel

File: packaging/spec.el7.in (hunk 5 above)
[the remark this answers is not in the document]
Comment (machacekondra) [reply packaging/spec.el7.in]: Ah, OK. So let's follow their guideline and ship only -devel package.

**Comment by imjoey** [comment]
> @imjoey It complains because you miss %files, you have only %files devel
> 
> ```
>     Empty %files file /home/jenkins/workspace/oVirt_ovirt-engine-sdk-go_standard-check-pr/ovirt-engine-sdk-go/rpmbuild/BUILD/ovirt-engine-sdk-go-4.3.0/debugfiles.list
> ```

Thanks for quick response. I was wondering why does el7 could be unaffected by this. Would adding a `%global debug_package %{nil}` in spec be proper to resolve this issue, as [Debuginfo](https://docs.fedoraproject.org/en-US/packaging-guidelines/Debuginfo/) said?

File: packaging/spec.el7.in [hunk 6]
```diff
@@ -0,0 +1,71 @@
+%global tar_version   4.3.0
+%global import_path   github.com/ovirt/go-ovirt
+%global gopath        %{_datadir}/gocode
+%global build_gopath  %{buildroot}/%{gopath}
+
+Name: go-ovirt-engine-sdk4
+Summary: Go SDK for version 4 of the oVirt Engine API
+Version: %{tar_version}
+Release: 2%{?dist}
```
Comment (machacekondra) [inline packaging/spec.el7.in]: Please replace 2 by 1.

File: packaging/spec.el7.in [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+%files devel
+%defattr(-,root,root,-)
+%doc README.md examples
+%license LICENSE.txt
+%dir %{gopath}/src/%{import_path}
+%{gopath}/src/%{import_path}/*.go
+
+
+%changelog
+* Tue Jan 29 2019 Joey Ma <majunjiev@gmail.com> - 4.3.0-2
```
Comment (machacekondra) [inline packaging/spec.el7.in]: 4.3.0-1

File: packaging/spec.fc28.in [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+%global tar_version   4.3.0
+%global import_path   github.com/ovirt/go-ovirt
+%global gopath        %{_datadir}/gocode
+%global build_gopath  %{buildroot}/%{gopath}
+
+Name: go-ovirt-engine-sdk4
+Summary: Go SDK for version 4 of the oVirt Engine API
+Version: %{tar_version}
+Release: 2%{?dist}
```
Comment (machacekondra) [inline packaging/spec.fc28.in]: Please change 2 to 1.

File: packaging/spec.fc28.in [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+%files devel
+%defattr(-,root,root,-)
+%doc README.md examples
+%license LICENSE.txt
+%dir %{gopath}/src/%{import_path}
+%{gopath}/src/%{import_path}/*.go
+
+
+%changelog
+* Tue Jan 29 2019 Joey Ma <majunjiev@gmail.com> - 4.3.0-2
```
Comment (machacekondra) [inline packaging/spec.fc28.in]: 4.3.0-1

File: packaging/spec.fc28.in (hunk 8 above)
Comment (imjoey) [reply packaging/spec.fc28.in]: Done, thanks.

File: packaging/spec.el7.in (hunk 7 above)
Comment (imjoey) [reply packaging/spec.el7.in]: Done, thanks.

File: packaging/spec.el7.in (hunk 6 above)
Comment (imjoey) [reply packaging/spec.el7.in]: Done, thanks.

File: packaging/spec.fc28.in (hunk 9 above)
Comment (imjoey) [reply packaging/spec.fc28.in]: Done, thanks.

**Pull request merged**